toolkit/committee-selection/authority-selection-inherents/src/lib.rs

```diff
@@ -111,7 +111,21 @@ impl<AuthorityId: Clone, AuthorityKeys: Clone> CommitteeMemberT
 }
 
 /// Trait to try extract [T] from [CandidateKeys].
-pub trait MaybeFromCandidateKeys: OpaqueKeys + Decode + Sized {
+///
+/// Chains using keys created using [impl_opaque_keys] should mark them with the [AutoMaybeFromCandidateKeys]
+/// and have their [MaybeFromCandidateKeys] automatically derived instead of implementing it manually.
+pub trait MaybeFromCandidateKeys: Sized {
+	/// Depends on `Decode` that is derived by `impl_opaque_keys!`
+	fn maybe_from(keys: &CandidateKeys) -> Option<Self>;
+}
+
+/// Marks candidate key types that should have their [MaybeFromCandidateKeys] instance automatically derived
+///
+/// Types for which auto-derivation is supported are:
+/// - types that implement [OpaqueKeys] trait, usually those created usig [impl_opaque_keys]
```

Comment:
We don't necessarily support types that have OpaqueKeys implementation. We depend on `impl_opaque_keys` macro.